Java/parkservices/src/main/java/com/amazon/randomcutforest/parkservices/threshold/ThresholdedRandomCutForest.java

```diff
@@ -222,7 +223,8 @@ private int maxContribution(DiVector diVector, int baseDimension, int startIndex
      * @return true/false if the residual (extrapolated) score would trigger anomaly
      *         designation
      */
-    protected boolean trigger(DiVector candidate, int gap, int baseDimension, DiVector ideal) {
+    protected boolean trigger(DiVector candidate, int gap, int baseDimension, DiVector ideal,
+            boolean previousIsPotentialAnomaly) {
```

Comment:
You mean just use the class variable and not pass through the function? We can - I thought it was better to create a function. One issue in this current refactor is that the grade computation is done by the other class and this class determines anomaly/not (because it corrects the grade, it only corrects in (anomaly) -> (not) direction)

Reply:
Yes, I meant to use the field previousIsPotentialAnomaly instead of the parameter previousIsPotentialAnomaly :) Not sure I understand the reasoning.  I generally use a field whenever possible.

Reply:
Fair point. There is a lot of state change going on in this code - it was easier to think of passing parameters which we would not accidentally change. Also there is a possibility that the computation of grade would be changed, but it is a possibility only at this point. Once the usage settles we can optimize this (and the basicThreshold() issue raised above -- that it was being called multiple times)